tensorflow_addons/metrics/r_square.py

```diff
@@ -138,12 +138,15 @@ def result(self) -> tf.Tensor:
             return tf.reduce_mean(raw_scores)
         if self.multioutput == "variance_weighted":
             return _reduce_average(raw_scores, weights=total)
-        raise RuntimeError(
```

Comment:
This is a redundant check as `result` is an method for `RSquare`, and it has been checked in init.